decisions/websre/0005-web-sre-kubernetes-secrets-pattern.md

```diff
@@ -0,0 +1,40 @@
+
+There are currently a few ways the secrets can be written to the cloud secret store. We have some terraform that writes the secrets directly to the secrets stores, but most secrets are populated manually by operators.  We expect both methods to be used going forward.  In general, we expect SREs/Operators with elevated privileges to manage the secrets.  This ADR intentionally doesn't include scope for devs to write new secrets, or update existing ones.  That problem is as of today, unsolved.
+
+The default assumption for storage patterns in aws secrets store is that you follow the pattern: `<group>/<app>/<environment>/<secretname>` for example `mozmeao/snippets/stage/database_url`  This seems to make writing aws iam access stanzas easy.  Not needing to be rewritten with each new secret added.  But, enough separation to prevent us from accidentally crossing wires.
+
+We are choosing to not use a few other types of secret storage mechanisms, including (but possibly not limited to):
+* git-crypt secret repos + sops, to k8s secrets directly.
+* Sealed secrets (which uses the kubernetes certs as an asymmetric key for local storage, and a crd in the cluster to present the secrets as env variables to the app)
+* SSM Parameter Store (lambdas/codebuilds use this)
+* no new systems to be built, for example hashicorp vault
+
+We expect CI/CD pipelines to use their current secret storage, and not try to connect them to the secret store.  The one exception is the cloud native pipeline solutions (e.g AWS CodeBuild), which should use the secrets manager if possible.  We may end up duplicating secrets, but in general we should try to keep application secrets and pipeline secrets separate, with pipeline secrets being manually added in the right context.
+
+## Consequences
+
+
+We will be migrating secrets, which is a reasonably risky process for many applications.
+We will have fewer methods for managing secrets, which ought to make it easier to rotate secrets, keep them up to date, etc.
+We are gatekeepers for adding new secrets, which may increase unplanned work for our team.
+If we were to lose the secrets manager completely, rebuilding would be a long and arduous process.
```

Comment:
To confirm my estimate of what this means, would it be equally as difficult as rotating every secret stored in the secrets manager, or more difficult because we would have lost access to the list of secrets?

Reply:
Probably a bit more difficult, there is some structure/path related data that glues the individual secrets together that we would also lose.  It's rotating every secret plus trying to get them all organized, both what path they live at and their internal structure (since secret objects can be key:value pairs, or just simple values).